frontend/device/input.lua

```diff
@@ -258,10 +259,21 @@ function Input:handleKeyBoardEv(ev)
         end
     end
 
-    if ev.value == EVENT_VALUE_KEY_RELEASE then
-        if keycode == "Light" then
-            return keycode
-        elseif keycode == "Power" then
+    if ev.value == EVENT_VALUE_KEY_RELEASE and keycode == "Light" then
+        return keycode
+    end
+    
+    if keycode == "Power" then
+        local UIManager = require("ui/uimanager")
+        if ev.value == EVENT_VALUE_KEY_PRESS then
+            power_up_trigerred = false
+            power_press_long = function()
+              power_up_trigerred = true
+              Device:onPowerEvent("PowerLong")
```

Comment:
Have you tried returning the string "PowerLong" directly here? You should be able to catch it later in `uimanager` by setting up `event_handlers` for PowerLong in `UIManager:init`. Then you don't have to import Device and UIManager. This way, you can also move `power_up_trigerred` and `power_press_long` into UIManager.